stl/inc/mdspan

```diff
@@ -307,18 +290,20 @@ public:
 
     constexpr mapping(const _Extents& _Ext) noexcept : _Myext(_Ext) {}
 
-    template <class _OtherExtents, enable_if_t<is_constructible_v<_Extents, _OtherExtents>, int> = 0>
+    template <class _OtherExtents>
+        requires is_constructible_v<_Extents, _OtherExtents>
     constexpr explicit(!is_convertible_v<_OtherExtents, _Extents>)
         mapping(const mapping<_OtherExtents>& _Other) noexcept
         : _Myext{_Other.extents()} {}
 
-    template <class _OtherExtents,
-        enable_if_t<_Extents::rank() <= 1 && is_constructible_v<_Extents, _OtherExtents>, int> = 0>
+    template <class _OtherExtents>
+        requires (_Extents::rank() <= 1) && is_constructible_v<_Extents, _OtherExtents>

```

Comment:
I observe that this PR is inconsistent about whether it refers to `_Extents::rank()` or `extents_type::rank()` in constraints. Similarly, it uses both `is_constructible_v<_Extents, _OtherExtents>` and `is_constructible_v<extents_type, _OtherExtents>`. This file should pick one form and use it consistently. (Ok to clean up in a following PR.)